utils/src/main/scala/za/co/absa/enceladus/utils/schema/SchemaUtils.scala

```diff
@@ -22,6 +22,45 @@ import scala.util.{Random, Try}
 
 object SchemaUtils {
 
+  /**
+   * Get a field from a text path and a given schema
+   * @param path   The dot-separated path to the field
+   * @param schema The schema which should contain the specified path
+   * @return       Some(the requested field) or None if the field does not exist
+   */
+  def getStructField(path: String, schema: StructType): Option[StructField] = {
+
+    @tailrec
+    def goThroughArrays(dataType: DataType): DataType = {
+      dataType match {
+        case ArrayType(dt, _) => goThroughArrays(dt)
+        case result => result
+      }
+    }
+
+    @tailrec
+    def diveInto(names: List[String], structField: StructField): Option[StructField] = {
```

Comment:
`goThroughArrays` and `diveInto` seem not consistent. Why not `goThroughStruct` and `diveIntoArrays`, for instance? (Althlough aestetically I like it, reads like a literature)
Maybe `processStruct` / `processArray`?